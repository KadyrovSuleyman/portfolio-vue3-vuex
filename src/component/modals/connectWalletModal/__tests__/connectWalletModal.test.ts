/* eslint-disable no-param-reassign */
import { mount, VueWrapper } from '@vue/test-utils';
import { createStore, Store } from 'vuex';
import Modal from '../connectWalletModal.vue';

jest.mock('../adapter.ts');
jest.mock('../walletsList/adapter.ts');

let store: Store<any>;
beforeEach(() => {
  store = createStore<any>({
    state: {
      isShown: false,
      hide: () => { store.state.isShown = false; },
    },
  });
});

let element: HTMLDivElement;
beforeEach(() => {
  element = document.createElement('div');
  element.className = 'app';
  document.body.appendChild(element);
});
afterEach(() => {
  document.body.outerHTML = '';
});

let wrapper: VueWrapper<any>;
afterEach(() => {
  wrapper.unmount();
});

it('connectWalletModal renders', async () => {
  wrapper = mount(Modal, {
    global: { plugins: [store] },
    props: {
      target: '.app',
    },
  });
  expect(element.outerHTML).toMatchSnapshot();

  store.state.isShown = true;
  await wrapper.vm.$nextTick();
  expect(store.state.isShown).toBeTruthy();
  expect(element.outerHTML).toMatchSnapshot();
});

it('closes', async () => {
  wrapper = mount(Modal, {
    global: { plugins: [store] },
    props: {
      target: '.app',
    },
  });

  expect(wrapper.findComponent('.connectWalletModal').exists()).toBeFalsy();
  expect(wrapper.findComponent('.background').exists()).toBeFalsy();

  store.state.isShown = true;
  await wrapper.vm.$nextTick();
  expect(store.state.isShown).toBeTruthy();

  wrapper.findComponent('.connectWalletModal-button').trigger('click');
  await wrapper.vm.$nextTick();
  expect(store.state.isShown).toBeFalsy();

  expect(wrapper.findComponent('.connectWalletModal').exists()).toBeFalsy();
  expect(wrapper.findComponent('.background').exists()).toBeFalsy();
});

it('watchs props changes', async () => {
  store.state.isShown = true;
  wrapper = mount(Modal, {
    global: { plugins: [store] },
    props: {
      target: '.app',
    },
  });
  expect(wrapper.findComponent('.connectWalletModal').classes()).toEqual(['connectWalletModal']);

  await wrapper.setProps({
    ...wrapper.props,
    mods: {
      selected: true,
    },
  });
  expect(wrapper.findComponent('.connectWalletModal').classes()).toEqual([
    'connectWalletModal',
    'connectWalletModal__selected',
  ]);

  await wrapper.setProps({
    ...wrapper.props,
    mods: {
      selected: false,
    },
  });
  expect(wrapper.findComponent('.connectWalletModal').classes()).toEqual(['connectWalletModal']);
});
