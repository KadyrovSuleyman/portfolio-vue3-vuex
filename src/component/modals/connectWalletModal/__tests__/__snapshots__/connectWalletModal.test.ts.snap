// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connectWalletModal renders 1`] = `
<div class="app">
  <transition-stub>
    <!--v-if-->
  </transition-stub>
  <transition-stub>
    <!--v-if-->
  </transition-stub>
</div>
`;

exports[`connectWalletModal renders 2`] = `
<div class="app">
  <transition-stub>
    <div class="connectWalletModal"><span class="connectWalletModal-header"> Select the payment card that you want to use for payment </span><span class="connectWalletModal-explanation"> The selected wallet will be connected to your staking </span>
      <div class="connectWalletModal-walletsList"></div><button class="connectWalletModal-button connectWalletModal-button__icon_cross"></button>
    </div>
  </transition-stub>
  <transition-stub>
    <div class="background"></div>
  </transition-stub>
</div>
`;
